### PR TITLE
feat: revamp currencies settings

### PR DESCRIPTION
**Pull request by kasperkristensen**
**What**
- Revamps the design and functionality of currency settings.
- Fixes bug where NextSelect was not re-rendered when options were changed.

**Note**
Searching in currency tables is missing from the implementation as it's not supported by the core atm.

Closes CORE-401

### CODE REVIEW

File: src/domain/settings/currencies/components/currency-tax-setting.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      includes_tax: currency.includes_tax,
+    })
+  }, [currency])
+
+  const onSubmit = handleSubmit((data: CurrencyTaxSettingFormType) => {
+    mutate(data, {
+      onSuccess: () => {
+        notification("Success", "Successfully updated currency", "success")
+      },
+      onError: (error) => {
```
Comment (fPolic) [inline src/domain/settings/currencies/components/currency-tax-setting.tsx:40]: suggestion: using `mutateAsync` will properly catch error, this way if we have 4xx the onError won't fire

File: src/domain/settings/currencies/components/currency-tax-setting.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        <Controller
+          control={control}
+          name="includes_tax"
+          render={({ field: { value, onChange } }) => {
+            return (
+              <Switch
+                checked={value}
+                onCheckedChange={(data) => {
+                  onChange(data)
+                  onSubmit()
```
Comment (fPolic) [inline src/domain/settings/currencies/components/currency-tax-setting.tsx:74]: also if there is an error on submit, we should revert toggle state

File: src/domain/settings/currencies/components/store-currencies/add-currencies-screen.tsx [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    useRowSelect
+  )
+
+  if (status === "error") {
+    return <div>Failed to load</div>
+  }
+
+  if (status === "loading" && !currencies) {
+    return <div>Loading...</div>
+  }
```
Comment (fPolic) [inline src/domain/settings/currencies/components/store-currencies/add-currencies-screen.tsx:91]: suggestion: can we add a bit of spacing around these messages
Screenshot 2022-09-15 at 10 58 35

**Review by fPolic — COMMENTED** [review]
Really nice work @kasperkristensen! I've added a couple of minor comments

File: src/domain/settings/currencies/components/currency-tax-setting.tsx (hunk 1 above)
Comment (kasperkristensen) [reply src/domain/settings/currencies/components/currency-tax-setting.tsx:40]: Why would it not catch the errors as is? Just tried it, and catches 400s without issue, is this not what you are experiencing?

File: src/domain/settings/currencies/components/currency-tax-setting.tsx (hunk 2 above)
Comment (kasperkristensen) [reply src/domain/settings/currencies/components/currency-tax-setting.tsx:74]: Definitely, I will add it asap 💪

File: src/domain/settings/currencies/components/currency-tax-setting.tsx (hunk 1 above)
Comment (fPolic) [reply src/domain/settings/currencies/components/currency-tax-setting.tsx:40]: Hmm, I get an uncaught error if it's 404

Comment (kasperkristensen) [reply src/domain/settings/currencies/components/currency-tax-setting.tsx:40]: Okay super interesting, can you give me some steps to try and reproduce? If I ensure a 404, it catches it properly and gives the correct error message in a notification.

image

**Review by olivermrbl — CHANGES_REQUESTED** [review]
This is great work @kasperkristensen - really neat user experience!

todo: TIP components should be hidden behind our feature flag component

File: src/domain/settings/currencies/components/currency-tax-setting.tsx (hunk 1 above)
Comment (kasperkristensen) [reply src/domain/settings/currencies/components/currency-tax-setting.tsx:40]: Okay, did some digging and it seems like `medusa-react` can't handle 404's that a thrown as a result of a route not existing, but handles "actual" 404's perfectly fine.

**Pull request merged**